### PR TITLE
add test and fix for kv grpc

### PR DESCRIPTION
**Pull request by ilovesoup**
Add test cases for tikv grpc calls.
Added exception propagate for non-grpc exceptions.

### CODE REVIEW

**Comment by ilovesoup** [comment]
Some catalog change slipped in. Please ignore.

File: src/test/com/pingcap/tikv/KVMockServer.java [hunk 1]
```diff
@@ -0,0 +1,227 @@
+package com.pingcap.tikv;
```
Comment (shenli) [inline src/test/com/pingcap/tikv/KVMockServer.java:16]: Add license

File: src/test/com/pingcap/tikv/RegionStoreClientTest.java [hunk 2]
```diff
@@ -0,0 +1,335 @@
+package com.pingcap.tikv;
```
Comment (shenli) [inline src/test/com/pingcap/tikv/RegionStoreClientTest.java:16]: Add license

File: src/main/java/com/pingcap/tikv/exception/KeyException.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.pingcap.tikv.exception;
```
Comment (shenli) [inline src/main/java/com/pingcap/tikv/exception/KeyException.java:16]: Add license

Comment (ilovesoup) [inline src/main/java/com/pingcap/tikv/exception/KeyException.java]: done

File: src/test/com/pingcap/tikv/RegionStoreClientTest.java (hunk 2 above)
Comment (ilovesoup) [reply src/test/com/pingcap/tikv/RegionStoreClientTest.java:16]: done

File: src/main/java/com/pingcap/tikv/exception/KeyException.java (hunk 3 above)
Comment (ilovesoup) [reply src/main/java/com/pingcap/tikv/exception/KeyException.java:16]: done

File: src/test/com/pingcap/tikv/KVMockServer.java (hunk 1 above)
Comment (ilovesoup) [reply src/test/com/pingcap/tikv/KVMockServer.java:16]: done

**Review by shenli — APPROVED** [review]
LGTM

**Pull request merged**